### PR TITLE
Add backend logic for skill cooldown HUD

### PR DESCRIPTION
**Pull request by veteran29**
- title

### CODE REVIEW

File: game/functions/core/client/displays/fn_displayAbilityCooldown.sqf [hunk 1]
```diff
@@ -16,37 +17,111 @@
...
+            };
+
+            // stop the loop
+            if (_remainingCooldown <= 0) exitWith {
+                _ctrlSeconds ctrlSetText "0 s";
+                _ctrlSeconds ctrlSetFade 1;
+                _ctrlSeconds ctrlCommit 3;
+
+                removeMissionEventHandler [_thisEvent, _thisEventHandler]
+            };
```
Comment (Spoffy) [inline game/functions/core/client/displays/fn_displayAbilityCooldown.sqf]: I think we could move this inside the `if` block above, and save ourselves from checking it every frame?

Comment (Spoffy) [reply game/functions/core/client/displays/fn_displayAbilityCooldown.sqf]: Or even:
```
if (_deltaT <= TICK_TIME) exitWith {
            _deltaT = _deltaT + diag_deltaTime;
            _thisArgs set [0, _deltaT];
}
```
And have the rest of the function run normally?

File: game/functions/systems/skills/client/active/fn_skills_active_assignSkillToSlot.sqf [hunk 2]
```diff
@@ -39,4 +39,10 @@ if (_skill isNotEqualTo createHashMap && {values vgm_c_skills_active_slots findI
     false
 };
 
-vgm_c_skills_active_slots getOrDefault [_slot, createHashMap] set ["skill", _skill] // return
+private _assigned = vgm_c_skills_active_slots getOrDefault [_slot, createHashMap] set ["skill", _skill];
+
+if (_assigned) then {
+    ["vgm_skills_active_slotted", [_slot, _skill]] call para_g_fnc_event_trigger;
```
Comment (Spoffy) [inline game/functions/systems/skills/client/active/fn_skills_active_assignSkillToSlot.sqf]: `triggerLocal` now, apologies for the inconveniently timed merge :D

File: game/functions/systems/skills/client/active/fn_skills_active_skillWheelActivate.sqf [hunk 3]
```diff
@@ -28,3 +30,5 @@ if (_slot call vgm_c_fnc_skills_active_isSlotOnCooldown) exitWith {
 player call (_skill get "codeActivate");
 private _cooldownUntil = time + (_skill get "cooldown");
 _slot set ["cooldownUntil", _cooldownUntil];
+
+["vgm_skills_active_activated", [_slot get "name", _skill]] call para_g_fnc_event_trigger;
```
Comment (Spoffy) [inline game/functions/systems/skills/client/active/fn_skills_active_skillWheelActivate.sqf]: `triggerLocal`

**Review by Spoffy — CHANGES_REQUESTED** [review]
Looks great. One suggestion, and two fixes for bugs caused by me merging the event system changes (sorry!)

File: game/functions/core/client/displays/fn_displayAbilityCooldown.sqf (hunk 1 above)
Comment (veteran29) [reply game/functions/core/client/displays/fn_displayAbilityCooldown.sqf]: Changed it so it's not checked on every frame and improved the timer accuracy a little bit by carrying over the `mod` over the ticker value upon ticker "tick".

**Review by Spoffy — APPROVED** [review]

**Pull request merged**